repos/system_upgrade/common/actors/scansourcefiles/tests/unit_test_scansourcefiles.py

```diff
@@ -1,5 +1,163 @@
+
+    file_info = scansourcefiles.scan_file(input_file)
+
+    assert isinstance(file_info, FileInfo)
+    assert file_info == expected_output
+
+
+@pytest.mark.parametrize('input_files,expected_output', (
+    ([], []),
+    (['file1'], [FileInfo(path='file1', exists=False, rpm_name='', is_modified=False)]),
+    (['file1', 'file2'], [FileInfo(path='file1', exists=False, rpm_name='', is_modified=False),
+                          FileInfo(path='file2', exists=False, rpm_name='', is_modified=False)]),
+))
+def test_scan_files(monkeypatch, input_files, expected_output):
+    def mocked_scan_file(input_file):
+        return FileInfo(path=input_file, exists=False, rpm_name='', is_modified=False)
+
+    monkeypatch.setattr(scansourcefiles, 'scan_file', mocked_scan_file)
+
+    assert scansourcefiles.scan_files(input_files) == expected_output
+
+
+@pytest.mark.parametrize('input_file,rpm_name,expected_output', (
+    (os.path.join(FILES_DIR, 'file_rpm_owned'), 'rpm',
+     FileInfo(path=os.path.join(FILES_DIR, 'file_rpm_owned'), exists=True, rpm_name='rpm', is_modified=False)),
+))
+def test_rpm_owned_files(monkeypatch, input_file, rpm_name, expected_output):
+    monkeypatch.setattr(api, 'produce', testutils.produce_mocked())
+    monkeypatch.setattr(scansourcefiles, 'get_source_major_version', lambda: 9)
+    monkeypatch.setattr(scansourcefiles, '_get_rpm_name', lambda _: rpm_name)
+    monkeypatch.setattr(scansourcefiles, 'is_modified', lambda _: False)
+    monkeypatch.setattr(scansourcefiles, 'TRACKED_FILES', {'common': [input_file]})
+
+    scansourcefiles.process()
+
+    tracked_files = api.produce.model_instances[0]
+    assert isinstance(tracked_files, TrackedFilesInfoSource)
+    file_info = tracked_files.files[0]
+    assert isinstance(file_info, FileInfo)
+
+    assert api.produce.called == 1
+    assert len(tracked_files.files) == 1
+    assert file_info == expected_output
+
+
+param_list = [(major_version,
+               FileInfo(path=os.path.join(FILES_DIR, 'file_major_version_{}'.format(major_version)),
+                        exists=False, rpm_name='', is_modified=False),
+               FileInfo(path=os.path.join(FILES_DIR, 'file_not_rpm_owned'),
+                        exists=True, rpm_name='', is_modified=False))
+              for major_version in RHEL_MAJOR_VERSIONS_LIST]
+
+
+@pytest.mark.parametrize('major_version,expected_output,common_expected_output', param_list)
+def test_version_file_with_existant_common_file(monkeypatch, major_version, expected_output, common_expected_output):
+    monkeypatch.setattr(api, 'produce', testutils.produce_mocked())
+    monkeypatch.setattr(api, 'current_logger', testutils.logger_mocked())
+    monkeypatch.setattr(scansourcefiles, 'get_source_major_version', lambda: major_version)
+    monkeypatch.setattr(scansourcefiles, '_get_rpm_name', lambda _: '')
+    monkeypatch.setattr(scansourcefiles, 'is_modified', lambda _: False)
```

Comment:
The same applies as described in the comment in `test_rpm_owned_files`. In this case it is valid to mock TRACKED_FILES as there are no files by version in the actor library, but again it is not required for the `file_not_rpm_owned` to physically exists (see the other comments).